istioctl/pkg/waypoint/waypoint.go

```diff
@@ -480,13 +481,24 @@ func deleteWaypoints(cmd *cobra.Command, kubeClient kube.CLIClient, namespace st
 			if gw.Spec.GatewayClassName != constants.WaypointGatewayClassName {
 				continue
 			}
-			names = append(names, gw.Name)
+			nameList = append(nameList, gw.Name)
+		}
+	} else {
+		for _, name := range names {
+			gw, err := kubeClient.GatewayAPI().GatewayV1().Gateways(namespace).Get(context.Background(), name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if gw.Spec.GatewayClassName != constants.WaypointGatewayClassName {
+				continue
```

Comment:
We can add an info message telling users why the gateway will not be deleted instead of silently skipping.

Reply:
Makes sense

Reply:
Updated